### PR TITLE
Speed up diagram generation

### PR DESCRIPTION
**Pull request by rodneykinney**
Some formatting changes also.

@schmmd FYI

### CODE REVIEW

File: src/main/scala/org/allenai/pipeline/Workflow.scala [hunk 1]
```diff
@@ -180,21 +183,21 @@ object Workflow {
   def forPipeline(steps: Iterable[(String, PipelineStep)], targets: Iterable[String], title: String, pipescripts: Option[PipescriptSources]): Workflow = {
     val idToName = steps.map { case (k, v) => (v.stepInfo.signature.id, k) }.toMap
     val nameToStep = steps.toMap
-    def findNodes(s: PipelineStep): Iterable[PipelineStep] =
-      Seq(s) ++ s.stepInfo.dependencies.flatMap {
-        case (name, step) =>
-          findNodes(step)
+    val allSteps = scala.collection.mutable.Set.empty[PipelineStep]
+    def findSteps(s: PipelineStep): Unit = {
```
Comment (rodneykinney) [inline src/main/scala/org/allenai/pipeline/Workflow.scala:187]: Change enumeration of nodes in the graph from N^2 to linear.

File: src/main/scala/org/allenai/pipeline/Workflow.scala [hunk 2]
```diff
@@ -72,14 +73,14 @@ case class Workflow(nodes: Map[String, Node], links: Iterable[Link], title: Stri
...
+                                                           |      labelType: "html",
+                                                           |      label: generateStepContent(${name.toJson},
+                                                                                                             |        ${desc.toJson},
+                                                                                                                                      |        ${info.executionInfo.toJson},
+                                                                                                                                                                             |        ${params.toJson},
+                                                                                                                                                                                                        |        ${linksJson})
+                                                                                                                                                                                                                               |    });""".stripMargin
```
Comment (dirkgr) [inline src/main/scala/org/allenai/pipeline/Workflow.scala:83]: What happened to the indentation here?

**Comment by dirkgr** [comment]
Other than the indentation, LGTM.

File: src/main/scala/org/allenai/pipeline/Workflow.scala [hunk 3]
```diff
@@ -180,21 +183,21 @@ object Workflow {
...
-    val nodeList = for {
-      name <- targets
-      step = nameToStep(name)
-      childStep <- findNodes(step)
-    } yield {
-      val id = childStep.stepInfo.signature.id
-      val childName = idToName.getOrElse(id, childStep.stepInfo.className)
-      (id, Node(childName, childStep))
     }
+    for (stepName <- targets) findSteps(nameToStep(stepName))
```
Comment (dirkgr) [inline src/main/scala/org/allenai/pipeline/Workflow.scala:193]: I prefer not being dogmatic about this stuff, but why didn't you stay pure and use a fold?

File: src/main/scala/org/allenai/pipeline/Workflow.scala (hunk 2 above)
Comment (rodneykinney) [inline src/main/scala/org/allenai/pipeline/Workflow.scala:83]: `sbt format` always does weird things to this code block. I've cleaned it up, but the formatter puts it in this state.

File: src/main/scala/org/allenai/pipeline/Workflow.scala (hunk 3 above)
Comment (rodneykinney) [inline src/main/scala/org/allenai/pipeline/Workflow.scala:193]: I don't think it maps cleanly to a fold because it's recursive on the nodes of the graph.

**Comment by dirkgr** [comment]
Are you making a release, so that `scholar` can depend on the new version?

**Comment by rodneykinney** [comment]
Yes, I'll make a release and submit a PR for scholar to use the new version.

**Pull request merged**